src/wasm2js.h

```diff
@@ -2574,16 +2598,35 @@ void Wasm2JSGlue::emitPostES6() {
   }
 }
 
-void Wasm2JSGlue::emitMemory(
-  std::function<std::string(std::string)> accessGlobal) {
+void Wasm2JSGlue::emitMemory() {
+  if (needsBufferView(wasm)) {
+    // Create a helper bufferView to access the buffer if we need one. We use it
+    // for creating memory segments if we have any (we may not if the segments
+    // are shipped in a side .mem file, for example), and also in bulk memory
+    // operations.
+    // This will get assigned during `asmFunc` (and potentially re-assigned
+    // during __wasm_memory_grow).
+    // TODO: We should probably just share a single HEAPU8 var.
+    out << "  var bufferView;\n";
+  }
+
   // If there are no memory segments, we don't need to emit any support code for
   // segment creation.
   if ((!wasm.memory.exists) || wasm.memory.segments.empty()) {
     return;
   }
 
+  // If we have passive memory segments, we need to store those.
+  for (auto& seg : wasm.memory.segments) {
+    if (seg.isPassive) {
+      out << "  var memorySegments = {};\n";
+      break;
+    }
+  }
+
   out <<
-    R"(for (var base64ReverseLookup = new Uint8Array(123/*'z'+1*/), i = 25; i >= 0; --i) {
+    R"(  var base64ReverseLookup = new Uint8Array(123/*'z'+1*/);
+  for (var i = 25; i >= 0; --i) {
```

Comment:
The old way is more compact (a single `var`).

Reply:
But it was horrendous to read.. I makes it look like `base64ReverseLookup` is scoped the for loop which it most certainly is not.  I don't think its worth the 4 bytes... if we want to go after 4 byte saving here I think there way much lower hanging fruit.

Reply:
If you feel strongly I'd rather find other vars i this file to collapse.

Reply:
For example couldn't we collapse all of these:

```
 var Math_imul = Math.imul;                                                      
 var Math_fround = Math.fround;                                                  
 var Math_abs = Math.abs;                                                        
 var Math_clz32 = Math.clz32;                                                    
 var Math_min = Math.min;                                                        
 var Math_max = Math.max;                                                        
 var Math_floor = Math.floor;                                                    
 var Math_ceil = Math.ceil;                                                      
 var Math_trunc = Math.trunc;                                                    
 var Math_sqrt = Math.sqrt;                                                      
 var abort = env.abort;   
```

?

Reply:
Fair enough, yeah, I don't feel strongly.

We should probably leave such micro-opts for closure anyhow. (which I think it does in closure=2)